### PR TITLE
Jupiter/poc/reinvite

### PR DESCRIPTION
**Pull request by kirill-konshin**

### CODE REVIEW

File: src/qos.ts [hunk 1]
```diff
@@ -62,6 +62,8 @@ export const startQosStatsCollection = (session: WebPhoneSession): void => {
 
     session.on('terminated', function() {
         previousGetStatsResult && previousGetStatsResult.nomore();
+        session.logger.log('Release media streams');
+        session && session.mediaStreams.release();
```
Comment (kirill-konshin) [inline src/qos.ts]: No need to check `session` here, you access `session.logger.log` 1 line above, or check there too

File: src/session.ts [hunk 2]
```diff
@@ -712,4 +723,46 @@ function addTrack(this: WebPhoneSession, remoteAudioEle, localAudioEle): void {
...
+    return false;
+}
+
+function stopMediaStats(session: WebPhoneSession) {
+    if (!session) {
+        return;
+    }
+    session.mediaStreams && session.mediaStreams.stopMediaStats();
+    session.mediaStatsStarted = false;
+    session.noAudioReportCount = 0;
```
Comment (kirill-konshin) [inline src/session.ts]: Shouldn't this be a method of session object?

File: src/session.ts [hunk 3]
```diff
@@ -712,4 +723,46 @@ function addTrack(this: WebPhoneSession, remoteAudioEle, localAudioEle): void {
...
+        return true;
+    }
+    if (!report.outboundRtpReport) {
+        return true;
+    }
+    if (report.inboundRtpReport.packetsReceived === 0 || report.outboundRtpReport.packetsSent === 0) {
+        return true;
+    }
+    return false;
+}
```
Comment (kirill-konshin) [inline src/session.ts]: This should reside in `src/rtpReport.ts`.

**Comment by kirill-konshin** [comment]
Also pls take a look at the amount of commits in the PR. I will squash and merge eventually, please re-sync your fork.

File: src/qos.ts (hunk 1 above)
Comment (jimmyxu1985) [reply src/qos.ts]: should check if session.mediaStreams exists

File: src/session.ts (hunk 2 above)
Comment (jimmyxu1985) [reply src/session.ts]: add stopMediaStats in type WebPhoneSession

File: src/session.ts (hunk 3 above)
Comment (jimmyxu1985) [reply src/session.ts]: moved into rtpReport.ts

**Comment by kirill-konshin** [comment]
Merged, please make sure you re-sync your fork.

**Pull request merged**